### PR TITLE
feat: expose `externalExperimentId` and `externalTrialId`

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-1632]

## Test Plan
- Check if `externalExperimentId` and `externalTrialId` are in static columns, and these are hidden by default in the new experiment list
- Check if  `externalExperimentId` and `externalTrialId` are filterable and sortable
- Check if `externalTrialId` is the `best_trial_id`

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-1632]: https://hpe-aiatscale.atlassian.net/browse/WEB-1632?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_experiment.go [hunk 1]
```diff
@@ -2195,7 +2197,9 @@ func sortExperiments(sortString *string, experimentQuery *bun.SelectQuery) error
 			WHERE t.experiment_id = e.id
 			ORDER BY searcher_metric_value_signed ASC
 			LIMIT 1
-		 ) `,
+		) `,
+		"externalExperimentId": "e.external_experiment_id",
+		"externalTrialId":      "t.external_trial_id",
```
Comment (gt2345) [inline master/internal/api_experiment.go:2202]: This should be `trials.external_trial_id`

Comment (keita-determined) [reply master/internal/api_experiment.go]: fixed

**Comment by gt2345** [comment]
Hi Keita, the changes look good to me, but there is a strange issue not sure if you can reproduce or if it's related. 
When I sort/pin/filter on a column, the two external id columns would disappear then re-appear after a while, I suspect this may have something to do with my local settings, but this still happens after I wiped all related settings from DB.

https://github.com/determined-ai/determined/assets/40620519/2d10caf3-ffce-4a72-b123-be75cf17fb24

**Comment by keita-determined** [comment]
> Hi Keita, the changes look good to me, but there is a strange issue not sure if you can reproduce or if it's related. When I sort/pin/filter on a column, the two external id columns would disappear then re-appear after a while, I suspect this may have something to do with my local settings, but this still happens after I wiped all related settings from DB.
> 
>  Screen.Recording.2023-09-11.at.4.47.11.PM.mov

@gt2345 Yes i noticed it too, but it is not from this PR. It happens to latest-main.

**Review by gt2345 — APPROVED** [review]
Changes look good to me

**Review by eecsliu — APPROVED** [review]
sql looks good to me!

**Pull request merged**